### DISCUSSION TITLE
404 on static files

### DISCUSSION THREAD

**Discussion by FuzzyGrim** [tiangolo/meinheld-gunicorn-docker#136] [Questions] [answered]
Hi, I have a django project and previously I was using uWSGI and nginx, I was serving static files directly with nginx. However, I wanted a single container for my project so I tried to migrate to this.

After setting it up with Django, everything seems to be working fine, except static files like css, js and images not loading. Do I need something like nginx to serve them as I did previously with this project as well?

From the docker logs I can see:
`Not Found: /static/css/main.css`

And if I enable debug mode, I get the standard Django error page saying:
```
 Using the URLconf defined in base.urls, Django tried these URL patterns, in this order:

1. [name='home'] 
2. ...

The current path, static/css/main.css, didn’t match any of these.
```

**Comment by tiangolo** [comment, marked as answer]
Hey! The uWSGI image with Nginx has some logic to make Nginx serve the static files directly. But this image is only Meinheld with Gunicorn serving your app. There's nothing handling your static assets, you would have to configure it in Django. I'm not sure what would be the right configuration for you, I guess it depends on your own code.

**Reply by FuzzyGrim** [reply]
Hi!

I see, thanks!
